src/pages/api/auth-connect/send-identity-validation-code.ts

```diff
@@ -33,84 +33,87 @@ import { getIdentityType } from "@src/utils/get-identity-type";
-      isCookieSealed: true,
-      cookieSalt: config.cookieSalt,
-    });
-
-    if (userCookie) {
-      const identity = {
-        type: getIdentityType(identityInput.type),
-        value: identityInput.value,
-      };
-
-      return await sendIdentityValidationCode(config.managementCredentials, {
-        callbackUrl,
-        identity,
-        localeCodeOverride: "en-EN",
-        userId: userCookie.sub,
-      })
-        .then(async ({ eventId }) => {
-          span.setDisclosedAttribute("is validation code sent", true);
-
-          let temporaryIdentity: TemporaryIdentity = {
-            eventId: eventId,
-            value: identityInput.value,
-            type: identityInput.type,
-            expiresAt: identityInput.expiresAt,
-            primary: identityInput.primary,
-          };
-
-          if (identityToUpdateId) {
-            temporaryIdentity = {
-              ...temporaryIdentity,
-              identityToUpdateId,
+  return tracer.span(
+    "re-send-identity-validation-code handler",
+    async (span) => {
+      const { callbackUrl, identityInput, identityToUpdateId } = request.body;
+
+      const userCookie = await getServerSideCookies<UserCookie>(request, {
+        cookieName: "user-cookie",
+        isCookieSealed: true,
+        cookieSalt: config.cookieSalt,
+      });
+
+      if (userCookie) {
+        const identity = {
+          type: getIdentityType(identityInput.type),
+          value: identityInput.value,
+        };
+
+        return await sendIdentityValidationCode(config.managementCredentials, {
+          callbackUrl,
+          identity,
+          localeCodeOverride: "en-EN",
+          userId: userCookie.sub,
+        })
+          .then(async ({ eventId }) => {
+            span.setDisclosedAttribute("is validation code sent", true);
+
+            let temporaryIdentity: TemporaryIdentity = {
+              eventId: eventId,
+              value: identityInput.value,
+              type: identityInput.type,
+              expiresAt: identityInput.expiresAt,
+              primary: identityInput.primary,
             };
-          }
 
-          await insertTemporaryIdentity(userCookie.sub, temporaryIdentity);
+            if (identityToUpdateId) {
+              temporaryIdentity = {
+                ...temporaryIdentity,
+                identityToUpdateId,
+              };
+            }
 
-          const verificationCodeMessage =
-            getIdentityType(identityInput.type) === IdentityTypes.EMAIL
-              ? "Confirmation email has been sent"
-              : "Confirmation SMS has been sent";
+            await insertTemporaryIdentity(userCookie.sub, temporaryIdentity);
 
-          setAlertMessagesCookie(response, verificationCodeMessage);
+            const verificationCodeMessage =
+              getIdentityType(identityInput.type) === IdentityTypes.EMAIL
+                ? "Confirmation email has been sent"
+                : "Confirmation SMS has been sent";
```

Comment:
May I suggest "A confirmation SMS/email has been sent", since in re-send, we display "A new ..." now ? 🙂

Reply:
Good point!